packages/material-ui/src/Modal/TrapFocus.js

```diff
@@ -113,7 +123,7 @@ function TrapFocus(props) {
         lastFocus.current = null;
       }
     };
-  }, [disableAutoFocus, disableEnforceFocus, disableRestoreFocus, isEnabled, open]);
+  }, [disableAutoFocus, disableEnforceFocus, disableRestoreFocus, open]);
```

Comment:
Previous to this change, providing a new `isEnabled` would break the trap logic.

Reply:
Could you add a test for this one? This seems off a bit. Usually removing a dependency shouldn't make changing the dependency work better.

Reply:
I believe that we will be able to significantly simplify the implementation in the future (-1kB gzipped):
https://github.com/facebook/react/blob/master/packages/react-events/docs/FocusScope.md

```jsx
import FocusScope from 'react-dom/FocusScope';

export default function TrapFocus(props) {
  const {
    children
    disableAutoFocus = false,
    disableEnforceFocus = false,
    disableRestoreFocus = false,
    open,
  } = props;

  if (!open) {
    return children;
  }

  return (
    <FocusScope
      autoFocus={!disableAutoFocus}
      contain={!disableEnforceFocus}
      restoreFocus={!disableRestoreFocus}
    >
      {children}
    </FocusScope>
  );
}
```

As long as the Modal tests are green, should we spend the effort on writing tests for it?

Reply:
I mean you can't change code without adding a test. Or maybe I missed the test added for this.

Reply:
Ok, I'm removing the changes from this pull request. It will be simpler, let's come back to it later.